daemon/image_service.go

```diff
@@ -60,6 +61,7 @@ type ImageService interface {
 	GetContainerLayerSize(ctx context.Context, containerID string) (int64, int64, error)
 	Mount(ctx context.Context, container *container.Container) error
 	Unmount(ctx context.Context, container *container.Container) error
+	Changes(ctx context.Context, container *container.Container) ([]archive.Change, error)
```

Comment:
For a follow-up, we must look into this signature.

When working on the CLI I was wondering why `pkg/archive` was imported in some location where I wouldn't expect it, and I stumbled upon this code; https://github.com/docker/cli/blob/14482589df194a86b2ee07df643ba3277b40df7d/cli/command/container/formatter_diff.go#L56-L67

```go
func (d *diffContext) Type() string {
	var kind string
	switch d.c.Kind {
	case archive.ChangeModify:
		kind = "C"
	case archive.ChangeAdd:
		kind = "A"
	case archive.ChangeDelete:
		kind = "D"
	}
	return kind
}
```

It turns out that the Client returns its own type for this, but that type does not have the enum values;
https://github.com/moby/moby/blob/1855a55d8c3cba353e6007f0ab355004ebe478d4/client/container_diff.go#L12
https://github.com/moby/moby/blob/1855a55d8c3cba353e6007f0ab355004ebe478d4/api/types/container/container_changes.go#L11

We should consider to either

- add the enums to the API types
- have the backend return the correct type
- have a utility to convert the type
- (possibly alias one for the other if needed)

Reply:
Opened a tracking ticket for the above;

- https://github.com/moby/moby/issues/45235